src/data-pipeline/etl-common/src/main/java/software/aws/solution/clickstream/common/sensors/SensorsEventParser.java

```diff
@@ -0,0 +1,341 @@
+        Map<String, ClickstreamEventPropValue> customParameters = new HashMap<>();
+        if (sensorsEvent.getUnknownProperties() != null) {
+            customParameters = convertStringObjectMapToStringEventPropMap(sensorsEvent.getUnknownProperties());
+            clickstreamEvent.setCustomParameters(customParameters);
+        }
+        String screenResolution = sensorsEvent.getProperties().getScreenWidth() + "x" + sensorsEvent.getProperties().getScreenHeight();
+        customParameters.put("distinct_id", new ClickstreamEventPropValue(sensorsEvent.getDistinctId(), ValueType.STRING));
+        customParameters.put("screen_resolution", new ClickstreamEventPropValue(screenResolution, ValueType.STRING));
+        customParameters.put("user_agent", new ClickstreamEventPropValue(sensorsEvent.getProperties().getUserAgent(), ValueType.STRING));
+
+        if (sensorsEvent.getProperties().getProvince() != null && sensorsEvent.getProperties().getCity() != null) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            ObjectNode eventObject = objectMapper.createObjectNode();
+            ObjectNode locationObject = objectMapper.createObjectNode();
+
+            locationObject.put("province", sensorsEvent.getProperties().getProvince());
+            locationObject.put("city", sensorsEvent.getProperties().getCity());
+
+            eventObject.set("event_location", locationObject);
+            customParameters.put("event_location",
+                    new ClickstreamEventPropValue(objectToJsonString(eventObject), ValueType.OBJECT));
+        }
+        if (sensorsEvent.getProperties() != null) {
+            customParameters.put("properties",
+                    new ClickstreamEventPropValue(objectToJsonString(sensorsEvent.getProperties()), ValueType.OBJECT));
+        }
+        return customParameters;
+    }
+
+    private static String mapEventName(final SensorsEvent sensorsEvent) {
+        return EVENT_NAME_MAP.getOrDefault(sensorsEvent.getEvent(), sensorsEvent.getEvent());
+    }
+
+    private ClickstreamUser getClickstreamUser(final SensorsEvent sensorsEvent, final ClickstreamEvent clickstreamEvent) throws JsonProcessingException {
+        ClickstreamUser clickstreamUser = new ClickstreamUser();
+        clickstreamUser.setAppId(clickstreamEvent.getAppId());
+        clickstreamUser.setEventTimestamp(clickstreamEvent.getEventTimestamp());
+        // clickstreamUser.setUserId(sensorsEvent.getUserId());
+        clickstreamUser.setUserPseudoId(clickstreamEvent.getUserPseudoId());
+
+        clickstreamUser.setFirstTouchTimeMsec(sensorsEvent.getTime());
+
+        if (clickstreamUser.getFirstTouchTimeMsec() == null) {
+            clickstreamUser.setFirstTouchTimeMsec(clickstreamEvent.getEventTimestamp().getTime());
+        }
+
+        clickstreamUser.setFirstVisitDate(new java.sql.Date(clickstreamUser.getFirstTouchTimeMsec()));
```

Comment:
Date() is deprecated. Use the new recommended method.

Reply:
It cannot be fixed in this scope, since clickstream model is using the sql.Date